proto/api.proto

```diff
@@ -0,0 +1,315 @@
+//          file_name = "" 
+//          block_id = 0
+//          bat= "test, 0, ..." 
+//      }
+
+// create a user table: 
+//  insert entry into mo_tables and ...;
+//      Entry {
+//          entry_type = Insert
+//          table_id = 1
+//          database_id =  0
+//          table_name =  mo_tables
+//          database_name = mo_catalog
+//          file_name = "" 
+//          block_id = 0
+//          bat.attrs = {"relid", "relname", "reldatabase", ...}
+//      }
+
+// drop table: 
+//  delete entry from mo_tables and ...;
+//      Entry {
+//          entry_type = delete
+//          table_id = 1
+//          database_id = 0
+//          table_name =  mo_tables
+//          database_name = mo_catalog
+//          file_name = "" 
+//          block_id = 0
+//          bat.attrs = {"relid", "relname",...}
+//      }
+
+//  bulk load:
+// bulk loads a block into S3.
+//      Entry {
+//          entry_type = Insert
+//          table_id = tid
+//          database_id = dbId
+//          table_name =  xxx
+//          database_name = xxx
+//          file_name = "s3 file name" 
+//          //transient block id;
+//          block_id = "xxx" 
+
+//          //block meta info and colunm meta info
+//          bat.attrs = {"clounm index", "min", "max", "bf location", "data location",...}
+//      }
+
+//  DML:
+// append a batch of data into table;
+//            Entry {
+    //          entry_type = Insert
+    //          table_id =  tid
+    //          database_id = dbId
+    //          table_name =  xxx
+    //          database_name = xxx
+    //          file_name = "" 
+    //          block_id = 0
+    //          bat.attrs = {"primary column value",  "column1 value", ...}
+    //      }
+
+//  delete batch of data from table;
+//            Entry {
+    //          entry_type = Insert
+    //          table_id =  tid
+    //          database_id = dbId
+    //          table_name =  xxx
+    //          database_name = xxx
+    //          file_name = "" 
+    //          block_id = 0
+    //          bat.attrs = {"rowid", "primary column value",  "column1 value", ...}
+    //      }
+
+
+message Entry {
+    enum EntryType {
+        Insert = 0;
+        Delete = 1;
+    }
+    EntryType entry_type = 1;
+    uint64    table_id = 2;
+    uint64    database_id = 3;
+    string    table_name = 4;
+    string    database_name = 5;
+    string    file_name = 6;
+    uint64    block_id = 7;
+    Batch     bat = 8;
+};
+
+// Checkpint is a snapshot at a timestamp for catalog and block info, 
+// it contains CatalogCkp and MetadataCkp.
+
+// CatalogCkp contains information about database and tables in the system,and
+// MetadataCkp contains information about blocks.
+message Checkpoint {
+    timestamp.Timestamp min_ts = 1;
+    timestamp.Timestamp max_ts = 2;
+    Batch      bat   = 3;
+
+};
+// catalog checkpoint:
+// one Batch represents a table, such as : mo_databases, mo_tables, mo_columns,... etc.
```

Comment:
Write this out in proto, not comment.   Let proto generate code, at least we know it is syntax wise correct.